servicetalk-traffic-resilience-http/src/main/java/io/servicetalk/apple/traffic/resilience/http/TrafficResilienceHttpClientFilter.java

```diff
@@ -0,0 +1,555 @@
+         * safe to be executed again (no previous invocation actually started) and because this maximizes the success
+         * chances.
+         * @param policy The {@link PeerCapacityRejectionPolicy} that represents the peer capacity rejection behavior.
+         * @return {@code this}.
+         */
+        public Builder peerCapacityRejection(final PeerCapacityRejectionPolicy policy) {
+            this.peerCapacityRejectionPolicy = requireNonNull(policy);
+            return this;
+        }
+
+        /**
+         * Peers can reject requests due to service unavailability.
+         * A {@link TrafficResilienceHttpClientFilter} can benefit from this input as feedback for the
+         * {@link CircuitBreaker} in use. A similar exception can be generated locally as a result of that feedback,
+         * to help the active local {@link CircuitBreaker} to also adapt.
+         * <p>
+         * It's important to know that if the passed {@code rejectionPredicate} tests {@code true} for a given
+         * {@link HttpResponseMetaData} then the operation will be {@link Single#failed(Throwable)}.
+         * <p>
+         * Out of the box if nothing custom is defined, the filter recognises as rejections requests with the following
+         * status codes:
+         * <ul>
+         *     <li>{@link HttpResponseStatus#SERVICE_UNAVAILABLE}</li>
+         * </ul>
+         *
+         * @param rejectionPredicate The {@link Function} that resolves a {@link HttpResponseMetaData response} to a
+         * peer-rejection or not.
+         * @return {@code this}.
+         */
+        public Builder peerUnavailableRejectionPredicate(
+                final Predicate<HttpResponseMetaData> rejectionPredicate) {
+            this.peerUnavailableRejectionPredicate = requireNonNull(rejectionPredicate);
+            return this;
+        }
+
+        /**
+         * When a peer rejects a {@link HttpRequestMetaData request} due to an open-circuit (see.
+         * {@link #peerUnavailableRejectionPredicate(Predicate)}), the feedback can be used
+         * to also forcefully open the local {@link HttpRequestMetaData request's} {@link CircuitBreaker}.
+         * The local {@link CircuitBreaker} will close again once a delay period passes as defined/extracted through
+         * the {@code delayProvider}.
+         * <p>
+         * If the delay provided is not a positive value, then the {@link CircuitBreaker} will not be modified.
+         *
+         * @param delayProvider A function to provide / extract a delay in milliseconds for the
+         * {@link CircuitBreaker} to remain open.
+         * @param executor A {@link Executor} used to re-close the {@link CircuitBreaker} once the delay expires.
+         * @return {@code this}.
+         */
+        public Builder forceOpenCircuitOnPeerCircuitRejections(
+                final Function<HttpResponseMetaData, Duration> delayProvider,
```

Comment:
Same question here

Reply:
discussed offline, there are dependent fields out of this actions, that fit better this API approach.

Reply:
Addressed in #2921